### ISSUE TITLE
Don't run watcher when using fuse-box bundle

### ISSUE THREAD

**Issue by RPDeshaies** [fuse-box/fuse-box-typechecker#6]
When I run my `build` task with the `TypeChecker` plugin, the process never return and my build never finishes.

My build task is using the `fuseBox.bundle('>main.ts');` instead of `fuseBox.devServer('>main.ts')`

I think the `TypeChecker plugin` should run only once when used with `fuseBox.bundle`.

**Comment by vegarringdal** [comment]
Ill have a look at during the week, I would expect it to only run once since it only run at bundle-end event, so it should have behaved the same if it was a bundle or devServer.

**Comment by vegarringdal** [comment]
@RPDeshaies Just checked it, its a bug, I need to run figure out what bundle is running and shut down the process if the bundle and not dev server is running

**Comment by RPDeshaies-playster** [comment]
Perfect !

**Comment by vegarringdal** [comment]
https://github.com/fuse-box/fuse-box/issues/409

**Comment by RPDeshaies** [comment]
Perfect !

**Comment by vegarringdal** [comment]
@RPDeshaies 

Added a temp fix for this ```TypeCheckPlugin({quit:true})``` if you add quit: true then it exits
Will improve this, but limited time atm, and guessing this was a annoying bug

**Comment by RPDeshaies** [comment]
Awesome. Thanks

**Comment by RPDeshaies** [comment]
I was asking this because I'm running the `TypeCheckPlugin` inside my Travis-CI.

With your `quit:true`, it works and the task is executed without problems.
The only problem is that it's not marking the build as `Failed`. 

Is it because the plugin should return some sort of error code when running with `quit: true` ?

**Comment by vegarringdal** [comment]
it prints fine here when I tested

![image](https://cloud.githubusercontent.com/assets/2901416/23962136/6eab5884-09ad-11e7-9574-53e86ba310e5.png)


I might have to do it all sync, maybe travis isnt picking up console messages running in a own thread

**Comment by RPDeshaies** [comment]
Sorry, I might have expressed myself badly.

Everything is printed. It's just not telling the CI that the build has failed because it has found errors

**Comment by vegarringdal** [comment]
Do you want it to throw error ?
I could prb add a option for that, tell me how you hope/think it should work and Ill have a look when I have better time

**Comment by RPDeshaies** [comment]
It would certainly be nice to throw errors that way it would be possible to refuse a PR if TypeCheck fails.

Except if you think its not the job of your plugin and the plugin should never be run on CI ?

Maybe I should use what the plugin is using directly ? What do you think about it ?

**Comment by vegarringdal** [comment]
We could add option for it to throw error, so that can be activated when needed.
Dont see any harm in having more options, most problematic part is time lately :-)
Plan was to have options for errors you do not care about too..

if we add a option to throw it should prb run before the bundler and maybe stop what ever output it causes, and option when to throw error, like semantic isnt anything that will make your app fail, usually..

**Comment by vegarringdal** [comment]
@RPDeshaies 

Im thinking 
```
({
  throwOnSyntactic: true,
  throwOnSemantic: true,
  throwOnGlobal: true,
  throwOnOtions: true,
  quit:true // need to be there until I get a way to figure out where it is
})
```

```throw 'type check errors'```

Or do did you have any ideas how you wanted it?

**Comment by RPDeshaies** [comment]
It seems perfect to me.

**Comment by vegarringdal** [comment]
@RPDeshaies tell me how the new option goes, not used travis, so guessing exit with code 1 triggers fail

**Comment by vegarringdal** [comment]
@RPDeshaies Did the changes I did help ?

**Comment by RPDeshaies** [comment]
I was not able to test it totally yet. 

I'll test it on monday ! But it seemed to work for now.

**Comment by vegarringdal** [comment]
How did the test go ?

**Comment by RPDeshaies** [comment]
Everything seems alright on my side ! :)

**Comment by vegarringdal** [comment]
updated this to support newer fusebox, see options/sample in readme.
Open a new issue if you see anything that does not work/ something Ive missed in the previews.
Will add more soon.

**Comment by vegarringdal** [comment]
@RPDeshaies 

Ive rebuild how it work
https://github.com/vegarringdal/fuse-box-typechecker

its not a plugin anymore, just a lib you can run in 3 ways
* sync , so you can check before you start fusebox
* async, paralell with build if wanted
*  watch, (works independent of fusebox, so you can use own cmd fro seeing errors, and have fusebox updates in own
